### PR TITLE
Tools: Testbench: Convert file component to module adapter

### PR DESCRIPTION
**Pull request by singalsu**
This is done as preparation for testbench IPC4 support. The update to IPC4 is simpler for a module adapter component.

### CODE REVIEW

**Comment by singalsu** [comment]
I've split out this from testbench IPC4 draft PR to ease review of large PR. I'll address the comments for file here.

File: tools/testbench/topology.c [hunk 1]
```diff
@@ -407,29 +432,39 @@ static int tb_register_filewrite(struct testbench_prm *tp,
...
+	file_uuid->b = 0x75aa;
+	file_uuid->c = 0x4ce8;
+	file_uuid->d[0] = 0x9d;
+	file_uuid->d[1] = 0xbe;
+	file_uuid->d[2] = 0xd8;
+	file_uuid->d[3] = 0xda;
+	file_uuid->d[4] = 0x08;
+	file_uuid->d[5] = 0xa6;
+	file_uuid->d[6] = 0x98;
+	file_uuid->d[7] = 0xc2;
```
Comment (singalsu) [inline tools/testbench/topology.c:457]: This was commented by Liam in #9025, is there a nicer way to set this with a macro?

Comment (lyakh) [reply tools/testbench/topology.c:457]: define a `static const` UUID and `memcpy()` it?

File: src/ipc/ipc3/helper.c [hunk 2]
```diff
@@ -211,20 +210,22 @@ static int comp_specific_builder(struct sof_ipc_comp *comp,
...
 	/* test bench maps host and DAIs to a file */
-	case SOF_COMP_HOST:
-	case SOF_COMP_SG_HOST:
-	case SOF_COMP_DAI:
-	case SOF_COMP_SG_DAI:
-		if (IPC_TAIL_IS_SIZE_INVALID(*file))
-			return -EBADMSG;
+	case SOF_COMP_FILEREAD:
+	case SOF_COMP_FILEWRITE:
 		config->file.channels = file->channels;
```
Comment (kv2019i) [inline src/ipc/ipc3/helper.c:215]: I think you need to keep the check on L218-219 (added recently).

**Review by kv2019i — COMMENTED** [review]
Look ok to  me, one comment inline

File: tools/testbench/topology.c (hunk 1 above)
Comment (singalsu) [reply tools/testbench/topology.c:457]: Yep, that works, thanks!

**Comment by singalsu** [comment]
There's a build fail in https://sof-ci.01.org/sof-pr-viewer/#/build/PR9432/build14219918, trying again.

**Comment by singalsu** [comment]
SOFCI TEST

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@wszypelt @lrudyX good for merge ?

**Comment by singalsu** [comment]
The Jenkins failures sof-ci/jenkins/pr-device-test/main-ace and sof-ci/jenkins/pr-device-test/main-lnl  are not related.  The Internal Intel CI System/merge/build that previously failed is now passed.

**Pull request merged**